internal/gengapic/testdata/custom_op_init.want

```diff
@@ -77,8 +77,7 @@ func NewRESTClient(ctx context.Context, opts ...option.ClientOption) (*Client, e
 	}
 	c.setGoogleClientInfo()
 
-	o := append(opts, option.WithHTTPClient(httpClient))
-	opC, err := NewFooOperationRESTClient(ctx, o...)
+	opC, err := NewFooOperationRESTClient(ctx, option.WithHTTPClient(httpClient))
```

Comment:
I forgot, do operations operate off the same endpoint as a mixin or are they global. If they are based off the endpoint the endpoint returned from creating the client should also be passed here.

Reply:
As in via `option.WithEndpoint`?

The operation client(s) for DIREGAPIC isn't standard LRO, but rather another API on the same endpoint defined in the same proto package. Thus, those operation clients by default target the same endpoint as the non-operation clients (those that precipitate the creation of operations). 

So both the containing client and the operation client refer to `compute.googleapis.com` and should refer to the same endpoint if changed at runtime. Hence we should also provide `WithEndpoint` using the endpoint from `httpClient` creation?

Reply:
And MTLS endpoint too?

Reply:
maybe not...

Reply:
Providing WithEndpoint too. Done.